bin/pre-commit-hook.js

```diff
@@ -46,16 +50,17 @@ const shouldFormat = text => {
 };
 
 // run prettier for any files in the commit that have @format within their first docblock
-files.map( file => path.join( __dirname, '../', file ) ).forEach( file => {
-	fs.readFile( file, 'utf8', ( err, text ) => {
+files
+	.map( file => path.join( __dirname, '../', file ) )
+	.forEach( file => {
+		const text = fs.readFileSync( file, 'utf8' );
```

Comment:
Right, this should fix it 👍

Reply:
We're losing the async part of this which doesn't need to be blocking. Probably not a big loss, but we could maintain the concurrency with some of the disk IO using something like the following:

```js
Promise.all(
	files.map( filename => {
		const file = path.join( __dirname, '../', filename );
		return new Promise( ( resolve, reject ) => {
			fs.readFile( file, 'utf8', ( err, text ) => {
				if ( err ) {
					reject( err );
				}
				if ( shouldFormat( text ) ) {
					console.log( `Prettier formatting file: ${ file } because it contains the @format flag` );
					const formattedText = prettier.format( text, {} );
					fs.writeFileSync( file, formattedText );
					execSync( `git add ${ file }` );
					resolve();
				}
			} );
		} );
	} )
);
```

☝️  _untested_